sphynx/networkit/additions.h

```diff
@@ -0,0 +1,11 @@
+// Additional code for the C++ side to simplify the interface.
+#include <vector>
+#include <networkit/Globals.hpp>
+#include <networkit/graph/Graph.hpp>
+
+struct EdgeList {
+  NetworKit::count n;
+  std::vector<unsigned long long> src, dst;
+};
+
+EdgeList graphToEdgeList(NetworKit::Graph& g);
```

Comment:
Aha, so all this is ours. But why do we need these two copy steps, isn't one enough? Is it not possible to pass over from go a simple pointer to a pre-allocated slice - hopefully looking like a `*unsigned long long` from he C++ side - and fill things up directly into the slice? You may need to first get `n` over to go in a separate call.

Reply:
> Aha, so all this is ours. But why do we need these two copy steps, isn't one enough? Is it not possible to pass over from go a simple pointer to a pre-allocated slice - hopefully looking like a `*unsigned long long` from he C++ side - and fill things up directly into the slice? You may need to first get `n` over to go in a separate call.

Done. I tried doing it with `SphynxId []` on the C++ side (instead of `SphynxId *`) but that refuses to work. Since I don't even understand why the current solution works, I better stop messing with it.